### PR TITLE
Enable gandiva on osx-arm, unpin PPC compilers

### LINKED ISSUE #849: Unpin very old GCC on PPC
I've tried to unpin the very old GCC on PPC in #848, but this runs into a myriad of errors of the following kind when trying to assemble `libgandiva.so`:
```
[clone .localalias]: error: relocation overflow
```

@hmaarrfk seemed to run into similar errors, hence the comment in the cbc.yaml

_Originally posted by @h-vetinari in https://github.com/conda-forge/arrow-cpp-feedstock/issues/848#issuecomment-1242753004_

### PR DESCRIPTION
**Pull request by h-vetinari**
Some follow-ups after the recent migrations

Closes #849

### CODE REVIEW

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
@pitrou 
I've tried to unpin the very old GCC on PPC, but this runs into a myriad of errors of the following kind when trying to assemble `libgandiva.so`:
```
[clone .localalias]: error: relocation overflow
```

Are you aware where this might be coming from or what the reason/solution might be?

**Comment by h-vetinari** [comment]
Also, it seems osx-arm is not yet ready to remove the almost 2 year old work-arounds for jemalloc/mimalloc...

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by xhochy** [comment]
> @pitrou I've tried to unpin the very old GCC on PPC, but this runs into a myriad of errors of the following kind when trying to assemble `libgandiva.so`:
> 
> ```
> [clone .localalias]: error: relocation overflow
> ```
> 
> Are you aware where this might be coming from or what the reason/solution might be?

This could be related to the `-fplt/-fno-plt` flag. We should look into this in a separate PR though

File: recipe/build-arrow.sh [hunk 1]
```diff
@@ -12,6 +12,10 @@ EXTRA_CMAKE_ARGS=""
 if [ "$(uname)" == "Linux" ]; then
   SYSTEM_INCLUDES=$(echo | ${CXX} -E -Wp,-v -xc++ - 2>&1 | grep '^ ' | awk '{print "-isystem;" substr($1, 1)}' | tr '\n' ';')
   EXTRA_CMAKE_ARGS=" -DARROW_GANDIVA_PC_CXX_FLAGS=${SYSTEM_INCLUDES}"
+else
+  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
+  EXTRA_CMAKE_ARGS=" -DARROW_CXXFLAGS=\"${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY\""
```
Comment (xhochy) [inline recipe/build-arrow.sh:17]: The escaping here is invalidated by the following line. Shouldn't it be sufficient to add `D_LIBCPP_DISABLE_AVAILABILITY` to CXXFLAGS directly?

Comment (h-vetinari) [reply recipe/build-arrow.sh:17]: I'm trying to sync with the [work](https://github.com/apache/arrow/commit/9d6598108c4fd93be89f8f5becaa7cb66f929fe7#diff-b81d923c288d441a9b33891d4cbc60323da63f1c906bdff67db1ff944296e834) from @pitrou upstream, but yeah, bash quoting rules got the better of me...

**Comment by xhochy** [comment]
> > This could be related to the `-fplt/-fno-plt` flag.
> 
> Yeah, that was the culprit in LLVM, but the errors looked quite different AFAIR.

I have seen quite a lot of different messages related to this in nodejs. I would first play around with this flag before investigating anything else.

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
> I have seen quite a lot of different messages related to this in nodejs. I would first play around with this flag before investigating anything else.

Good guess, that was the solution! 🥳 

Meanwhile, I've figured out osx-arm + gandiva too, so this PR can basically unfold its full formerly intended glory 😋

**Review by xhochy — APPROVED** [review]
Nice

**Comment by h-vetinari** [comment]
> Nice

Better double-check what I just force-pushed a few seconds before your review, but I'm confident it is identical.

**Comment by h-vetinari** [comment]
OK, there was no change here for linux-aarch, and the rest is green (or failing spuriously), so merging. 🙃

**Pull request merged**